File: ex_happy-primes/submissions/mvds-c/happyprime.c
h(c,C,r,p){for(;C>1&&C%++p;);for(;c;c/=10)r+=c%10*(c%10);r&~5?h(r,C,0,1):printf(
"%s %sprime",r-1?"sad":"happy",p>=C&C>1?"":"non-");}main(c){h(c,c,0,scanf("%d",&c));}

File: ex_happy-primes/submissions/seth-py/happyprime.py
import re,sys
def ge(z):
 while z!=1:z=sum((int(a)**2 for a in str(z)));yield z
def h(n):
 l=[];g=ge(n)
 while True:
  try:z=g.next()
  except:return True
  if z in l:return False
  l.append(z)
def p(n):return re.match(r'^1?$|^(11+?)\1+$','1'*n)is None
n=int(sys.argv[1])
print "%s %sprime"%(('sad','happy')[h(n)],('non-','')[p(n)])
